runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -10217,7 +10554,10 @@ static TR::Register *inlineConcurrentLinkedQueueTMOffer(TR::Node *node, TR::Code
    static char * disableTMOffer = feGetEnv("TR_DisableTMOffer");
    static char * debugTM = feGetEnv("TR_DebugTM");
 
-   if (disableTMOffer)
+   /*
+    * TODO: TM is currently not compatible with read barriers. If read barriers are required, TM is disabled until the issue is fixed.
+    */
+   if (disableTMOffer || TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads())
```

Comment:
I doubt this is worth the investment. It also won't work with hardware CS. See #3763 for an explanation.

Reply:
What do you mean by investment? We disabled TM, so there is no investment.

Reply:
Do you mean we just need to update the comment?

Reply:
I believe the issue here is the comment claims this is something we might fix and enable in the future. It seems like it might not be worth it to fix and I should just change the comment to simply say TM is not compatible with read barriers and that it gets disabled if read barriers are needed and not have it be a TODO.